cumulusci/cli/cci.py

```diff
@@ -809,6 +809,13 @@ def org_connect(config, org_name, sandbox, login_url, default, global_org):
     oauth_dict = oauth_capture()
     org_config = OrgConfig(oauth_dict, org_name)
     org_config.load_userinfo()
+    org_config._load_orginfo()
+    if not org_config.organization_sobject["TrialExpirationDate"]:
+        org_config.config["expiration"] = None
+    else:
+        org_config.config["expiration"] = org_config.organization_sobject[
+            "TrialExpirationDate"
+        ]
```

Comment:
In the case of ScratchOrgConfig, there is a Python property called `expires` which returns a datetime object (or maybe it's a date? I forget) calculated from the `date_created` and the `days`.

I think we should aim to have both types of org config return something similar when you access `org_config.expires`. So that means we should parse the TrialExpirationDate into a date or datetime and store it in `expires` instead of `expiration`.